contrib/docker/Dockerfile

```
@@ -1,12 +1,18 @@
-FROM dealii/dependencies:focal
+ARG IMG=focal
+
+FROM dealii/dependencies:$IMG
 
 LABEL maintainer="luca.heltai@gmail.com"
 
+ARG VER=master
+
 USER root
-RUN cd /usr/src \
-    && git clone https://github.com/dealii/dealii \
-    && cd dealii && mkdir build && cd build \
-    && cmake -GNinja \
+RUN cd /usr/src && \
+    git clone https://github.com/dealii/dealii dealii-$VER && \
+    cd dealii-$VER && \
```

```suggestion
    git clone https://github.com/dealii/dealii && \
    cd dealii && \
```

We ship each image with only one version of deal.II, right? To keep all images in the same structure, I would suggest to keep the filenames of the source directories the same as well.

I like the `VER` argument though! It's an easy way to switch between branches/versions.

---

We only ever use the source directory when debugging. The actual library is installed in /usr/local for all images, and that's all we really need to know about.  I think there is value in distinguishing one version from the other and to allow multiple versions of the source, but if you feel strongly about this, I'll make the change you suggest.

---

No, I don't feel strongly about it.